packages/devtools/lib/src/timeline/cpu_profile_model.dart

```diff
@@ -194,10 +201,40 @@ class CpuStackFrame extends TreeNode {
     return _inclusiveSampleCount;
   }
 
+  CpuStackFrame shallowCopy({bool copySampleCounts = false}) {
+    final copy = CpuStackFrame(
+      id: id,
+      name: name,
+      category: category,
+      url: url,
+      profileTime: profileTime,
+      profileSampleCount: profileSampleCount,
+    );
+    if (copySampleCounts) {
+      copy
+        ..exclusiveSampleCount = exclusiveSampleCount
+        .._inclusiveSampleCount = _inclusiveSampleCount;
+    }
+    return copy;
+  }
+
+  /// Returns a deep copy from this stack frame down to the leaves of the tree.
+  ///
+  /// The returned copy stack frame will have a null parent.
+  CpuStackFrame deepCopy() {
+    final copy = shallowCopy(copySampleCounts: true);
+    for (CpuStackFrame child in children) {
+      copy.addChild(child.deepCopy());
+    }
+    return copy;
+  }
+
+  bool shallowEquals(CpuStackFrame other) =>
```

Comment:
nit: maybe name this in a way that makes it clearer why I would use shallowEquals?

Reply:
renamed to `matches` and added an explanatory comment.